integration_tests/README.md

```diff
@@ -5,41 +5,161 @@ are intended to be able to be run against any Spark-compatible cluster/release t
+  [here](https://github.com/pyenv/pyenv#installation)
+- Verify that `pyenv` is set correctly
+  
+  ```shell script
+  which pyenv  
+  ```
+
+- Using `pyenv` to set Python installation
+  - To check versions to be installed (will return a long list)
+  
+    ```shell script
+    ls ~/.pyenv/versions/
+    ```
+
+  - To install a specific version from the available list
+  
+    ```shell script
+    pyenv install 3.X.Y
+    ```
+
+  - To check available versions locally
+
+    ```shell script
+    ls ~/.pyenv/versions/
+    ```
+
+  - To set python environment to one of the installed versions
+
+    ```shell script
+    pyenv global 3.X.Y
+    ```
+
+For full details on `pyenv` and instructions, see [pyenv github page](https://github.com/pyenv/pyenv).
+</p>
+</details>
+
+<details><summary><b>Installing specific version of maven</b></summary>
+<p>
+
+All package managers like `brew` and `apt` offers maven. However, it may lag behind some
+versions. In that case, you can install the latest binary from the [Maven download page](https://maven.apache.org/download.cgi).
+For manual installation, you need to setup your environment:
+
+  ```shell script
+  export M2_HOME=PATH_TO_MAVEN_ROOT_DIRECTOTY
+  export M2=${M2_HOME}/bin
+  export PATH=$M2:$PATH
+  ```
+
+</p>
+</details>
+
+### Dependencies
+
+- pytest
+  : A framework that makes it easy to write small, readable tests, and can scale to support complex
+  functional testing for applications and libraries (requires  Python 3.6+).
+- sre_yield
+  : Provides a set of APIs to generate string data from a regular expression.
+- pandas
+  : A fast, powerful, flexible and easy to use open source data analysis and manipulation
+  tool and is only needed when testing integration with pandas.
+- pyarrow
+  : Provides a Python API for functionality provided by the Arrow C++ libraries, along with
+  tools for Arrow integration and interoperability with pandas, NumPy, and other software in
+  the Python ecosystem. This is used to test improved transfer performance to pandas based user
+  defined functions.
+- pytest-xdist
+  : A plugin that extends pytest with new test execution modes, the most used being distributing
+  tests across multiple CPUs to speed up test execution
+- findspark
+  : Adds pyspark to sys.path at runtime
+
+You can install all the dependencies using `pip` by running the following command:
+
+  ```shell script
+  pip install pytest \
+              sre_yield \
+              pandas \
+              pyarrow \
+              pytest-xdist \
+              findspark
+  ```
+
+### Installing Spark
+
+You need to install spark-3.x and set `SPARK_HOME/bin` to your `$PATH`.
+
+To build a distribution run the following command. Full instructions can be found [here](https://spark.apache.org/docs/latest/building-spark.html).
+
+To run the plugin against the latest spark, you will need to buid a new version from source:
+
+  ```shell script
+  ## to build a distribution from branch is branch-3.x.y
```

Comment:
Spark does not use a patch version for branches, it's just `branch-x.y` (e.g.: Spark 3.2.2-SNAPSHOT source is on `branch-3.2`).
```suggestion
  ## The source for Spark snapshot version X.Y.Z is on branch-x.y
  ```